pkgs/development/python-modules/chart-studio/default.nix

```diff
@@ -3,53 +3,57 @@
   buildPythonPackage,
   fetchFromGitHub,
   mock,
-  nose,
   plotly,
-  pytest,
   requests,
   retrying,
   setuptools,
-  six,
+  pytestCheckHook,
 }:
 
 buildPythonPackage rec {
   pname = "chart-studio";
-  version = "5.22.0";
+  version = "5.23.0-unstable-2024-07-23";
   pyproject = true;
 
   # chart-studio was split from plotly
   src = fetchFromGitHub {
     owner = "plotly";
     repo = "plotly.py";
-    rev = "refs/tags/v${version}";
-    hash = "sha256-cEm0vLQ4PAVxvplqK+yayxLpNCvyfZtjZva0Bl2Sdfs=";
+    # Get version by splitting the version by dashes and getting first element.
+    # Should always be the x.x.x version number.
+    rev = "refs/tags/v${lib.findFirst (v: v == v) version (lib.splitString "-" version)}";
+    hash = "sha256-K1hEs00AGBCe2fgytyPNWqE5M0jU5ESTzynP55kc05Y=";
```

Comment:
Given my requested `version` change, you can just inline the tag version directly here, or `let plotlyVersion = …;`, or even just use `plotly.version` (but I wouldn’t necessarily recommend that; moving `plotly` over to `fetchFromGitHub` and inheriting its entire `src` could work, though).

Reply:
Moving plotly over to fetchFromGitHub is a larger task than I want to do in this pr as it needs NPM dependencies. Let's merge this and revisit later if this is really a big issue. I'll update to use the chart-studio version in the public version number but plotly's version number for the git tag.

Reply:
I agree that it’s probably best not to couple the sources for now. However, in that case we probably shouldn’t inherit from `plotly`’s version, because the FOD hash won’t be updated when `plotly` does, causing confusing and nondeterministic results depending on what previous versions you’ve built. I think we should just hardcode the tag for now.

Reply:
Tag hardcoded, we can revisit this when/if plotly gets moved to fetchFromGitHub